Automattic/wp-calypso
# Stats: Refactor local state updates in date picker

## Proposed Changes

Cleans up the logic inside the DateControlPicker to make things a bit easier to read/maintain.

**Note:** Based on work done in #83157 so that should be merged first.

## Testing Instructions



1. Open Live branch to the **Stats → Traffic** page.
2. Enable date picker: `?flags=stats/date-control`
3. Confirm that shortcuts continue to work correctly, mainly noting that the date inputs update as needed.

## Pre-merge Checklist
